sdk/core/azure-core-http-netty/src/main/java/com/azure/core/http/netty/NettyAsyncHttpClientBuilder.java

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.http.netty;
+
+import com.azure.core.http.ProxyOptions;
+import com.azure.core.util.logging.ClientLogger;
+import io.netty.channel.nio.NioEventLoopGroup;
+import reactor.netty.http.client.HttpClient;
+import reactor.netty.tcp.ProxyProvider;
+
+import java.util.function.Supplier;
+
+/**
+ *
+ */
+public class NettyAsyncHttpClientBuilder {
+    private final ClientLogger logger = new ClientLogger(NettyAsyncHttpClientBuilder.class);
+
+    private Supplier<ProxyOptions> proxyOptions;
+    private boolean enableWiretap;
+    private int port;
+    private NioEventLoopGroup nioEventLoopGroup;
+
+    /**
+     *
+     */
+    public NettyAsyncHttpClientBuilder() {    }
+
+    /**
+     *
+     * @return A new NettyAsyncHttpClient instance
+     * @throws IllegalStateException If proxy type is unknown.
+     */
+    public NettyAsyncHttpClient build() {
+        HttpClient nettyHttpClient = HttpClient.create()
+            .port(port)
```

Comment:
@alzimmermsft Can you please validate that this is sane when the user does not specify a port, and therefore the port value is zero. Thanks!